exercises/starknet/basics/starknet1.cairo

```diff
@@ -0,0 +1,27 @@
+// starknet1.cairo
+// Starkling, Joe, is writing a really simple contract.
+// The contract shows that he is the owner of the contract.
+// However, his contract is not working. What's he missing?
+
+// I AM NOT DONE
+
+#[contract]
+mod JoesContract {
+
+    fn get_owner() -> felt252 {
+        'Joe!'
+    }
+
+}
+
+#[cfg(test)]
+mod test {
+    use array::ArrayTrait;
+    use array::SpanTrait;
+    use super::JoesContract;
+    #[test]
+    #[available_gas(2000000000)]
+    fn test_contract_view() {
+        JoesContract::__external::get_owner(ArrayTrait::new().span());
```

Comment:
Your tests should not use the function defined under the `__external` namespace! If you're unit testing, you have to call `JoesContract::get_owner`.
If you're testing against deployed contracts, you would call `joes_contract.get_owner()`

This should be modified in the following files as well.

Reply:
Starknet contract tests in `starkware-libs/cairo` repo seem to use `__external`...

Reply:
the result is the same and in our case it makes it _a lot_ simpler to read and understand

Reply:
Yeah, but if down the road, someone unintentionally deletes/mistypes the attributes then tests would pass but...
So I thought using the `::external...` would be nice...

Reply:
This is specially important for exercises where we are teaching them how to write contracts with working endpoints.
So a decided to go with this complication in tests.

Reply:
I'm strongly against it

Reply:
But wouldn't the exercise pass even if they miss(type) the annotations. We don't want that.

You are welcome to improve them by writing deployment for class and testing that if you prefer. Maybe a helper util func that deploys and returns deployment to test?

Reply:
> I'm strongly against it

Removed all `__external` calls.